cli/src/semgrep/commands/scan.py

```diff
@@ -512,19 +513,32 @@ def convert(
     ),
     optgroup.group("Semgrep Pro Engine options"),
     optgroup.option(
-        "--pro-languages",
-        is_flag=True,
-        help="Enable Pro languages (currently just Apex). Requires Semgrep Pro Engine, contact support@r2c.dev for more information on this.",
+        "--pro",
+        "requested_engine",
+        type=EngineType,
+        flag_value=EngineType.PRO_INTERFILE,
+        help="Inter-file analysis and Pro languages (currently just Apex). Requires Semgrep Pro Engine, contact support@r2c.dev for more information on this.",
     ),
     optgroup.option(
         "--pro-intrafile",
-        is_flag=True,
+        "requested_engine",
+        type=EngineType,
+        flag_value=EngineType.PRO_INTRAFILE,
         help="Intra-file inter-procedural taint analysis. Implies --pro-languages. Requires Semgrep Pro Engine, contact support@r2c.dev for more information on this.",
     ),
     optgroup.option(
-        "--pro",
-        is_flag=True,
-        help="Inter-file analysis and Pro languages (currently just Apex). Requires Semgrep Pro Engine, contact support@r2c.dev for more information on this.",
+        "--pro-languages",
+        "requested_engine",
+        type=EngineType,
+        flag_value=EngineType.PRO_LANG,
+        help="Enable Pro languages (currently just Apex). Requires Semgrep Pro Engine, contact support@r2c.dev for more information on this.",
+    ),
+    optgroup.option(
+        "--oss-only",
```

Comment:
Note @brandonspark `--oss-only` was not a feature in `semgrep scan` before but will now be. Probably fine unless this was intentional